Resources/Public/Javascript/PageView/ImageManipulationControl.js

```diff
@@ -8,435 +8,235 @@
  * LICENSE.txt file that was distributed with this source code.
  */
 
-/**
- * Right know the image manipulation uses an own ol.Map object based on a webgl renderer. This is due to the fact
- * that other parts of the viewer application are using vector geometries and ol3 does only support full vector
- * renderering with the canvas and dom renderer yet. In contrast the image manipulation tool is only working
- * with a webgl renderer. Therefore it uses an own ol.Map object which is overlaid and synchronized with the
- * base ol.Map object.
- *
- * @constructor
- * @param {Object=} options Control options.
- *  {Array.<ol.layer.Layer>} layers
- *  {Element} target
- *  {ol.View} view
- *  {ol.Map} map
- */
-dlfViewerImageManipulationControl = function(options) {
-
-    /**
-     * @type {Object}
-     * @private
-     */
-    this.dic = $('#tx-dlf-tools-imagetools').length > 0 && $('#tx-dlf-tools-imagetools').attr('data-dic') ?
-        dlfUtils.parseDataDic($('#tx-dlf-tools-imagetools')) :
-        {'imagemanipulation-on':'Activate image manipulation', 'imagemanipulation-off':'Deactivate image manipulation', 'saturation':'Saturation', 'hue':'Hue', 'brightness':'Brightness', 'contrast':'Contrast', 'reset':'Reset', 'invert':'Color inverting', 'parentContainer':'.tx-dlf-imagemanipulationtool'};
-
+class ImageManipulationControl extends ol.control.Control {
     /**
-     * @type {Array.<ol.layer.Layer>}
-     * @private
+     * @param {Options=} optOptions Image Manipulation options.
      */
-    this.layers = options.layers;
+    constructor(optOptions) {
+        const options = optOptions ? optOptions : {};
+
+        super({
+            element: document.createElement('div'),
+            target: options.target,
+        });
+
+        const className = options.className !== undefined ? options.className : 'ol-custom-image-manipulation';
+        const cssClasses = className + ' ' + ol.css.CLASS_UNSELECTABLE + ' ' + ol.css.CLASS_CONTROL;
+
+        const autoOpen = options.autoOpen !== undefined ? options.autoOpen : false;
+        const label = options.label !== undefined ? options.label : '\u25d1';
+        const tipLabel = options.tipLabel !== undefined ? options.tipLabel : 'Toggle image manipulation';
+        const title = options.title !== undefined ? options.title : 'Image Manipulation';
+
+        const button = document.createElement('button');
+        button.title = tipLabel;
+        button.setAttribute('type', 'button');
+        button.appendChild(typeof label === 'string' ? document.createTextNode(label) : label);
+        button.addEventListener('click', this, false);
+
+        const element = this.element;
+        element.className = cssClasses;
+        element.appendChild(button);
+
+        /**
+         * @type {Object}
+         * @private
+         */
+        this.filterDefaults_ = {
+            contrast: 1,
+            saturate: 1,
+            brightness: 1,
+            'hue-rotate': '0turn',
```

Comment:
Why hue-rotate filter is inside the ' ' and others not?

Reply:
Otherwise the `-` would be interpreted as operator.